linode/manager
# upcoming: [M3-7220, M3-7355] - Add DC Get Well logic to various user flows pt2 and cleanup

## Description 📝
- Adds some additional logic to disable regions when creating kubernetes
- cleans up existing DC Get Well logic (make a prop required, clean up comments) + update tests as needed

## Preview 📷
kubernetes msw
![image](https://github.com/linode/manager/assets/139280159/1a259f97-7cb9-430f-80f4-9fccf77b90c2)

## How to test 🧪

### Prerequisites
### Verification steps 
Without the MSW (and with/without the DC get well feature flag), for Kubernetes and VPCs
- Verify that the creation (or migration) flow for each entity still works as expected
- Verify that this PR still has the same regions as prod
  - ** Tokyo no longer appears in this PR bc we've removed fake regions
  - ** order of continent/regions may not be the same, we've alphabetized everything

With the MSW + dc get well feature flag, for Kubernetes only
- Verify that disabled regions appear as expected for each entity/capability (see mock data for account/availability endpoint in serverHandlers.ts)
- Verify that if a region doesn't have a capability for some entity, that region doesn't appear. You can do this by adding or removing capabilities in regions.json
- clicking on the "learn more" link takes you to https://www.linode.com/global-infrastructure/availability (page does not yet exist but will soon)

## As an Author I have considered 🤔

*Check all that apply*
